source/custom/nixnet_service.custom.cpp

```diff
@@ -1,2 +1,168 @@
+      }
+      case i32_: {
+        i32 property_value{};
+        status = library_->GetProperty(session_ref, property_id, property_size, &property_value);
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        response->set_i32_scalar(property_value);
+        break;
+      }
+      case f64_: {
+        f64 property_value{};
+        status = library_->GetProperty(session_ref, property_id, property_size, &property_value);
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        response->set_f64_scalar(property_value);
+        break;
+      }
+      case string_: {
+        std::string property_value = "";
+        property_value.resize(property_size, '0');
+        status = library_->GetProperty(session_ref, property_id, property_size, property_value.data());
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        response->set_str(property_value.c_str());
+        break;
+      }
+      case u32_array_: {
+        response->mutable_u32_array()->mutable_u32_array()->Clear();
+        response->mutable_u32_array()->mutable_u32_array()->Resize(property_size, 0);
+        u32* property_value = reinterpret_cast<u32*>(response->mutable_u32_array()->mutable_u32_array()->mutable_data());
+        status = library_->GetProperty(session_ref, property_id, property_size, property_value);
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        break;
+      }
+      case ref_: {
+        u32 property_value{};
+        status = library_->GetProperty(session_ref, property_id, property_size, &property_value);
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        response->mutable_ref()->set_id(property_value);
+        break;
+      }
+      case string_array_: {
+        std::string property_value = "";
+        property_value.resize(property_size, '0');
+        status = library_->GetProperty(session_ref, property_id, property_size, property_value.data());
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        response->set_string_array(property_value.c_str());
+        break;
+      }
+      case ref_array_: {
+        std::vector<u32> property_value(property_size, 0);
+        status = library_->GetProperty(session_ref, property_id, property_size, property_value.data());
+        if (!status_ok(status)) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        auto response_iterator = response->mutable_ref_array()->mutable_ref()->begin();
+        for (auto it = property_value.begin(); it != property_value.end(); it++, response_iterator++) {
```

Comment:
This is not correct. Since this is a session, you need to `access_session` to get the grpc session from the db_ref.

Also: is it possible that the db_ref is not yet tracked by grpc-device? @doshirohan 's change handles that for `FindDBRef` by basically making it an `init_method`. The only challenge there is that we need to provide an array of session_names so that the user can name the sessions.

You might want to defer this to a separate task and make sure you know how this is used.

Reply:
I have added a logic for both ref and ref_array type propertyids, could you pl have a look at it?